File: Cpp/300.cpp
//直观解法
//动态规划
//时间复杂度O（n^2）
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int len = nums.size();
        vector<int> dp(len,1);
        for(int i=0; i<len; i++) {
            for(int j=0; j<i; j++) {
                if(nums[i] > nums[j]) {
                    dp[i] = max(dp[i],dp[j]+1);
                }
            }
        }
        int res = dp[0];
        for(int i=1; i<len; i++) {
            res = max(res, dp[i]);
        }
        return res;
    }
};

File: JavaScript/300.js
// dp
// 时间 n^2 空间 n
var lengthOfLIS = function(nums) {
    let dp = [1], max = 1; //nums[i]结尾的最长递增子序列
    for(let i=1; i<nums.length; ++i) {
        dp[i] = 1;
        for(let j=0; j<i; ++j) {
            if(nums[j] < nums[i]) {
                dp[i] = Math.max(dp[j] + 1, dp[i]);
            }
        }
        max = Math.max(dp[i], max);
    }
    return max;
};